src/test/java/test/epam/learn/webdriver/hurtmeplenty/page/CloudPricingCalculatorPage.java

```diff
@@ -0,0 +1,185 @@
+    WebElement estimateInstanceType;
+
+    @FindBy(xpath = "//*[@id='compute']//md-list-item[4]")
+    WebElement estimateRegion;
+
+    @FindBy(xpath = "//*[@id='compute']//md-list-item[5]")
+    WebElement estimateLocalSSD;
+
+    @FindBy(xpath = "//*[@id='compute']//md-list-item[6]")
+    WebElement estimateCommitmentTerm;
+
+    @FindBy(xpath = "//*[@id='resultBlock']//h2/b")
+    WebElement totalEstimatedCost;
+
+    public CloudPricingCalculatorPage(WebDriver driver) {
+        super(driver);
+    }
+
+    public CloudPricingCalculatorPage newForFrame() {
+        driver.switchTo().frame(iframe).switchTo().frame(myFrame);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForNumberOfInstances(String number) {
+        numberOfInstances.sendKeys(number);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForOperatingSystem() {
+        operatingSystemButton.click();
+        clickAsJavascriptExecutor(selectedOperatingSystem);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForVMClass() {
+        clickAsJavascriptExecutor(vMClassButton);
+        clickAsJavascriptExecutor(selectedVMClass);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForInstanceType() {
+        clickAsJavascriptExecutor(instanceTypeButton);
+        clickAsJavascriptExecutor(selectedInstanceType);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForAddGPUs() {
+        clickAsJavascriptExecutor(addGPUsBox);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForNumberOfGPUs() {
+        clickAsJavascriptExecutor(numberOfGPUsButton);
+        clickAsJavascriptExecutor(selectedNumberOfGPUs);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForGPUType() {
+        clickAsJavascriptExecutor(GPUTypeButton);
+        clickAsJavascriptExecutor(selectedGPUType);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForLocalSSD() {
+        clickAsJavascriptExecutor(localSSDButton);
+        clickAsJavascriptExecutor(selectedLocalSSD);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForDatacenterLocation() {
+        clickAsJavascriptExecutor(datacenterLocationButton);
+        clickAsJavascriptExecutor(selectedDatacenterLocation);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForCommitedUsage() {
+        clickAsJavascriptExecutor(commitedUsageButton);
+        clickAsJavascriptExecutor(selectedCommitedUsage);
+        return this;
+    }
+
+    public CloudPricingCalculatorPage newForAddToEstimate() {
+        clickAsJavascriptExecutor(addToEstimateButton);
+        return this;
+    }
+
+    public String findEstimateVMClass() {
+        return estimateVMClass.getText();
+    }
+
+    public String findEstimateInstanceType() {
+        return estimateInstanceType.getText();
+    }
+
+    public String findEstimateRegion() {
```

Comment:
Are you really think that getText() method is finding something? It just returns the text from the element. Please update naming.